rbtcwrapperproxy/RBTCWrapperProxy.sol

```diff
@@ -123,6 +158,82 @@ contract RBTCWrapperProxy{
+     * 4.Calls 'addLiquidity' on LiquidityPoolConverter contract
+     * 5.Transfers pool tokens to user
+     * 
+     * @param _liquidityPoolConverterAddress    address of LiquidityPoolConverter contract
+     * @param _reserveTokens                    address of each reserve token. The first element should be the address of WRBTC
+     * @param _reserveAmounts                   amount of each reserve token. The first element should be the amount of RBTC
+     * @param _minReturn                        minimum return-amount of reserve tokens
+     *
+     * @return amount of pool tokens minted
+     */
+    function addLiquidityToV1(
+        address _liquidityPoolConverterAddress,
+        IERC20Token[] memory _reserveTokens, 
+        uint256[] memory _reserveAmounts, 
+        uint256 _minReturn
+    )  
+        public
+        payable
+        checkAddress(_liquidityPoolConverterAddress)
+        returns(uint256) 
+    {
+        require(address(_reserveTokens[0]) == wrbtcTokenAddress, "The first reserve token must be WRBTC");
+        require(_reserveAmounts[0] == msg.value, "The provided amount of RBTC should be identical to msg.value");
+
+        bool success;
+
+        IWrbtcERC20(wrbtcTokenAddress).deposit.value(_reserveAmounts[0])();
+        success = IWrbtcERC20(wrbtcTokenAddress).approve(_liquidityPoolConverterAddress, _reserveAmounts[0]);
+        require(success, "Failed to approve converter to transfer WRBTC");
+        
+        for (uint256 i = 1; i < _reserveTokens.length; i++) {
+            IERC20Token reserveToken = _reserveTokens[i];
+            uint256 reserveAmount = _reserveAmounts[i];
+            success = IERC20Token(reserveToken).transferFrom(msg.sender, address(this), reserveAmount);
+            require(success, "Failed to transfer reserve token from user");
+            success = IERC20Token(reserveToken).approve(_liquidityPoolConverterAddress, reserveAmount);
+            require(success, "Failed to approve converter to transfer reserve token");
+        }
+
+        ILiquidityPoolV1Converter _liquidityPoolConverter = ILiquidityPoolV1Converter(_liquidityPoolConverterAddress);
+        ISmartToken _poolToken = ISmartToken(address(_liquidityPoolConverter.token()));
+        uint256 poolTokenAmountBefore = _poolToken.balanceOf(address(this));
+        _liquidityPoolConverter.addLiquidity(_reserveTokens, _reserveAmounts, _minReturn);
+        uint256 poolTokenAmount = _poolToken.balanceOf(address(this)).sub(poolTokenAmountBefore);
+        
+        success = _poolToken.transfer(msg.sender, poolTokenAmount);
+        require(success, "Failed to transfer pool token to user");
+
+        (uint256 rsvBalance, , , , ) = _liquidityPoolConverter.reserves(wrbtcTokenAddress);
+        uint256 costReserveAmount = ISovrynSwapFormula(addressOf(SOVRYNSWAP_FORMULA)).fundCost(_poolToken.totalSupply(), rsvBalance, _liquidityPoolConverter.reserveRatio(), poolTokenAmount);
```

Comment:
The parameters are not entirely correct. You are passing the total supply, reserve balance and ratio after the liquidity was added, while the converter is using the total supply, reserve balance and ratio before the liquidity was added as input for the fund cost calculation.

the ratio shouldn't change in the meantime, so I wouldn't expect this making any difference, but I consider it safer to use the same parameters the liquidity converter used.